Fixes #15171. Improved the Optional Arguments doc

Dawny33 opened the pull request:
Updated the Optional arguments doc, to make it more beginner friendly, as discussed in this issue: (https://github.com/JuliaLang/julia/issues/15171)

innerlee commented on doc/manual/functions.rst:
```diff
@@ -413,29 +413,25 @@ Optional Arguments
 
-    julia> parse(Int,"12")
-    12
+    julia> parse(2,3,4)
```
myfunc?

Dawny33 replied:
@innerlee Thank You. Updated the commit :+1:

nalimilan commented on doc/manual/functions.rst:
```diff
@@ -413,29 +413,25 @@ Optional Arguments
-library function :func:`parse(type,num,base) <parse>` interprets a string as a number
-in some base. The ``base`` argument defaults to ``10``. This behavior can be
+might not need to be passed explicitly in every call. For example, consider the
+function myfunc(x,y, z=1). The ``z`` argument defaults to ``1``. This behavior can be
```
Use double backquotes `` around code blocks. Also, probably better call the function `f` for consistency with examples below.

nalimilan replied:
Ah, and always add a space before commas.

nalimilan commented on doc/manual/functions.rst:
```diff
@@ -413,29 +413,25 @@ Optional Arguments
 
 With this definition, the function can be called with either two or three
-arguments, and ``10`` is automatically passed when a third argument is not
+arguments, and ``1`` is automatically passed when a third argument is not
```
Having three arguments is not necessary (I guess that was because `parse` has three of them). Using just `x` and `y=1` should be enough.

Dawny33 replied:
@nalimilan Initially built a PR around two arguments, however a function with three arguments looked more intuitive. So, went ahead with it.

nalimilan commented:
Thanks! This looks good to me apart from the comments I made. Could you squash everything into one commit once you'll have addressed them, so that this can be merged?

Dawny33 commented:
@nalimilan Does it look good enough now?

Commit:
Updated the function name and fixed formatting for improved readability

The example is now copy-paste'able and easier to follow and understand for new comers.

nalimilan commented:
Thanks, looks good, except for the commit message, which should mention that it affects the section of the manual about optional arguments; it should also say "Fixes #15171" and explain the reason for this change.

JeffBezanson commented on doc/manual/functions.rst:
```diff
@@ -413,29 +413,25 @@ Optional Arguments
-    julia> parse(Int,"12",3)
-    5
+    julia> f(2, 3)
+    2,3,1
```
The actual output has parentheses.

Dawny33 replied:
@JeffBezanson Thanks for pointing it out! Updated with the latest commit.

JeffBezanson commented:
I disagree that phony `f`, `foo`, and `bar` functions are "easier to understand" than an actual example. I think it makes the manual quite boring and tiresome when every example is contrived.

Commit:
Updated the example with correct output

Dawny33 commented:
@JeffBezanson Yeah, I agree with you that regular function names makes it quite boring and tiresome. 

However, the problem which this PR solves, is to make the example(the function in the example) copy-pastable, according to the issue here (https://github.com/JuliaLang/julia/issues/15171), so that it does not cause confusion for newbies. In the current docs, `type` which is reserved, is used as an argument for the `parse` function.  (Corresponding discussion in Gitter)[https://gitter.im/JuliaLang/julia?at=56c963a35c3b0559674da3e0]

The function name can always be changed.

tkelman commented:
@Dawny33 could you reopen this pull request against the master branch? We prefer to have changes go there first (then we can backport them separately) rather than be made against the release branch first, in which case master will not get the same change.

JeffBezanson commented:
@Dawny33 Yes, but that's easy to fix without resorting to everybody's favorite letters f, x, y, and z. In fact, it would be better to find other places in the manual where we are using `f` or `foo` and replace them with real, useful examples.

Dawny33 commented:
@tkelman Done. Pl refer #15321. Let me know if I can close this PR.

@JeffBezanson Yeah, maybe we should!  

However, including better examples which can be easily reproducible in the REPL (and/or) in a script, instead of examples having `reserved` terms, etc should be a priority, as they help starters avoid unnecessary confusion :)

omus commented:
I think it would be good to still include the `parse` usage examples.

tkelman commented:
Let's close this in favor of #15321 for now. Once that has been revised and merged we can look at reopening and updating this for the backport, or I'll just include it in a future rollup.
